### PR TITLE
Complete picoquant driver for basic working

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/driver/picoquant.py [hunk 1]
```diff
@@ -343,6 +525,131 @@ def ReadFiFo(self, count):
...
+
+        self._metadata[model.MD_BINNING] = (b, 1)
+        self._metadata[model.MD_PIXEL_DUR] = self.GetResolution() * 1e-9  # ps -> s
+
+        return (b, 1)
+
+    def _setSyncOffset(self, offset):
+        offset_ps = int(offset * 1e9)
+        self.SetSyncOffset(offset_ps)
+        offset = offset_ps * 1e-9
```
Comment (ktsitsikas) [inline src/odemis/driver/picoquant.py]: Is it needed to recalculate it?

File: src/odemis/driver/picoquant.py [hunk 2]
```diff
@@ -343,6 +525,131 @@ def ReadFiFo(self, count):
...
+                        if state == "T":
+                            return
+                    except Queue.Empty:
+                        pass
+
+                # Keep acquiring
+                while True:
+                    tacq = self.dwellTime.value
+                    tstart = time.time()
+                    tend = tstart + tacq * 3 + 1
```
Comment (ktsitsikas) [inline src/odemis/driver/picoquant.py]: How is that calculated?

File: src/odemis/driver/picoquant.py [hunk 3]
```diff
@@ -343,6 +525,131 @@ def ReadFiFo(self, count):
...
+                                elif state == "T":
+                                    return
+                            except Queue.Empty:
+                                pass
+
+                            # Is the data ready?
+                            if self.CTCStatus():
+                                logging.debug("acq still running")
+                                break
+                        else:
```
Comment (ktsitsikas) [inline src/odemis/driver/picoquant.py]: will this ever happen? Shouldn't now get updated within the while loop?

**Comment by pieleric** [comment]
All the comments should now be addressed

**Pull request merged**